### PR TITLE
Removes coupons outdated banner as well as its associated survey

### LINKED ISSUE #12956: Remove outdated coupons banner from coupon list screen
iOS counterpart issue: https://github.com/woocommerce/woocommerce-ios/issues/14467

Coupons feature has been enabled for more than a year now. The current banner from the screen feels outdated and doesn't make sense anymore. Also the banner is shown when the search icon is tapped. 

https://github.com/user-attachments/assets/48e96398-fbdb-4aa3-83df-d2c255a11474

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #12956 

### Description

Coupons feature has been released for more than a year now (close to 2 years). The banner with the feature announcement and the link to the survey felt outdated and didn't make sense anymore. We decided to remove it. 

This PR does just that. Removes the banner and cleans the code for any reference related to the banner or the coupons survey. 

### Testing information
1. Make a clean install of the app. 
2. Open More Menu > Coupons section
3. Check the banner is not displayed anymore and that coupons list is displayed correctly. 

### The tests that have been performed

The above. Additionally I tested that the empty case for coupons is also displayed correctly. 

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - [Internal] Refactored IPP Payment flow to allow customizing payment collection UI in POS [https://github.com/woocommerce/woocommerce-android/pull/13014]
 - [*] Blaze Campaign Intro screen now offers creating a new product if the site has no products yet [https://github.com/woocommerce/woocommerce-android/pull/13001]
 - [*] When entering a wrong WordPress.com account for login, retrying will bring the step back to the email input screen [https://github.com/woocommerce/woocommerce-android/pull/13024]
+- [*] Removes coupons feature announcement banner [https://github.com/woocommerce/woocommerce-android/pull/13077]
```
Comment (hichamboushaba) [inline RELEASE-NOTES.txt]: np, should we mark this as [Internal]?, to me this is not something that we want to communicate to users in a release note, WDYT?

**Review by hichamboushaba — APPROVED** [review]
Thanks for taking care of this @JorgeMucientes, code looks good.

I left on np comment, but it's not a blocker, I'm pre-approving.

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (JorgeMucientes) [reply RELEASE-NOTES.txt]: True! Makes sense. Marked as internal

**Pull request merged**